matching-notification-ui/src/main/resources/MatchingNotification/MatchesTable.xml

```diff
@@ -127,20 +129,44 @@ $xwiki.jsx.use('PhenoTips.Widgets')##
+                  $escapetool.xml($services.localization.render('phenotips.matchingNotifications.table.matchesSeletFilter.local'))&lt;span class="matches-count"&gt;&lt;/span&gt;##
+                  #set ($matchesUpdateObject = $matchesRunDoc.getObject($infoClassname, 'serverName', 'local'))##
+                  #if ($matchesUpdateObject)##
+                    #set($endTime = $matchesUpdateObject.getValue('completedTime'))##
+                  #end##
+                  &lt;input type="hidden" class="disclaimer" value="$escapetool.xml($services.localization.render('phenotips.myMatches.disclaimer.disclaimer.PhenomeCentral'))"&gt;##
+                  &lt;input type="hidden" class="endTime" value="${escapetool.xml($!endTime)}"&gt;##
+                  &lt;input type="hidden" class="serverName" value="$escapetool.xml($services.localization.render('phenotips.matchingNotifications.table.matchesSeletFilter.local'))"&gt;##
+                  &lt;span class="fa fa-info-circle mme-disclaimer-help" title="$escapetool.xml($services.localization.render('phenotips.myMatches.disclaimer.showDisclaimer'))"&gt;&lt;/span&gt;
+                  &lt;br&gt;
                 &lt;/label&gt;
                 &lt;div class="across-mme"&gt;&lt;input id="mme-filter" type="checkbox" checked &gt; $escapetool.xml($services.localization.render('phenotips.myMatches.filters.mme.label')) &lt;a href="http:\/\/matchmakerexchange.org" target="_blank"&gt;MME&lt;/a&gt;:&lt;/div&gt;
                 #set ($preferenceDoc = $xwiki.getDocument("XWiki.XWikiPreferences"))
                 #set ($allRemotes = $preferenceDoc.getObjects("PhenomeCentral.RemoteMatchingServiceConfiguration"))
                 #foreach ($config in $allRemotes)
                   #if ($config.getProperty('serverId').getValue() != "" &amp;&amp; $config.getProperty('humanReadableName').getValue() != "" &amp;&amp; $config.getValue('includeInUI') == 1)
-                    &lt;label&gt;&lt;input name="checkbox-server-id-filter" class="mme" type="checkbox" value="$config.getProperty('serverId').getValue()" checked&gt;
-                      &lt;div class="filter-mme-server-name"&gt;$escapetool.xml($config.getProperty('humanReadableName').getValue())&lt;span class="matches-count"&gt;&lt;/span&gt;&lt;/div&gt;&lt;br&gt;
+                    #set ($serverID = $config.getProperty('serverId').getValue())
+                    #set ($serverName = $config.getProperty('humanReadableName').getValue())
+                    #if ($serverName == "")
+                      #set ($serverName = $serverID)
+                    #end
+                    #set ($disclaimer = $config.getProperty('disclaimer').getValue())
+                    #set ($matchesUpdateObject = $matchesRunDoc.getObject($infoClassname, 'serverName', $serverID))
+                    #if ($matchesUpdateObject)
+                      #set($endTime = $matchesUpdateObject.getValue('completedTime'))
```

Comment:
If the object is null, `$endTime` is not updated and retains its previous value. If matches were computed on the first remote but not on the second, the second remote will report that it had its matches computed at the same time as the first.

```suggestion
                      #set ($endTime = $matchesUpdateObject.getValue('completedTime'))
                   #else
                     #set ($endTime = '')
```

or even better, make it shorter, skip the `$matchesUpdateObject` assignment altogether and ensure the `$endTime` is always updated, with just one line:

```
#set ($endTime = "$!matchesRunDoc.getObject($infoClassname, 'serverName', $serverID).getValue('completedTime')")
```